### PR TITLE
Issue #2644 create different namespace for each template of xpaas

### PR DESCRIPTION
**Pull request by amitkrout**
Fix of issue(s) : #2644

NOTE : Using Make file changes for my testing propose, later will revert back the changes before merge. 

How to test :

1. make integration GODOG_OPTS=“-tags addon-xpaas”
2.

### CODE REVIEW

**Comment by amitkrout** [comment]
Retest this please

**Comment by amitkrout** [comment]
Retest this please

**Comment by amitkrout** [comment]
Retest this please

**Comment by amitkrout** [comment]
Retest this please

**Comment by amitkrout** [comment]
100% success rate in ci. 5 success out of 5 try. Reverting back the Makefile changes and removing [WIP] tag. @minishift/minishift-dev Pr is open for review

File: test/integration/features/addon-xpaas.feature [hunk 1]
```diff
@@ -19,18 +19,18 @@ which are then available in OpenShift to the user.
 
   Scenario Outline: User deploys, checks out and deletes several templates from XpaaS imagestream
    Given Minishift has state "Running"
-    When executing "oc new-app <template-name>" succeeds
-     And executing "oc set probe dc/<service-name> --readiness --get-url=http://:8080<http-endpoint>" succeeds
```
Comment (anjannath) [inline test/integration/features/addon-xpaas.feature]: We don't need to set a readiness probe anymore?

**Comment by anjannath** [comment]
CI looks good :+1:

File: test/integration/features/addon-xpaas.feature (hunk 1 above)
Comment (amitkrout) [reply test/integration/features/addon-xpaas.feature]: In a confusion i removed it. In the next commit i will add it.

File: test/integration/features/addon-xpaas.feature [hunk 2]
```diff
@@ -19,18 +19,18 @@ which are then available in OpenShift to the user.
...
+    When executing "oc new-project <project-name>" succeeds
+    Then executing "oc new-app <template-name>" succeeds
```
Comment (LalatenduMohanty) [inline test/integration/features/addon-xpaas.feature]: -1, we should not remove the `oc set` command as it creates the  probe

File: test/integration/features/addon-xpaas.feature (hunk 1 above)
Comment (anjannath) [reply test/integration/features/addon-xpaas.feature]: @amitkrout You can update this PR and add back the readiness probe.

**Comment by amitkrout** [comment]
Retest this please

File: test/integration/features/addon-xpaas.feature (hunk 1 above)
Comment (amitkrout) [reply test/integration/features/addon-xpaas.feature]: Done

File: test/integration/features/addon-xpaas.feature (hunk 2 above)
Comment (amitkrout) [reply test/integration/features/addon-xpaas.feature]: Done, added back

**Comment by amitkrout** [comment]
screen shot 2018-08-09 at 7 09 41 pm

I added back ```readiness probe```. Again the success rate is 100% with 3 tries

@LalatenduMohanty Pr is ready for merge

File: test/integration/features/addon-xpaas.feature [hunk 3]
```diff
@@ -19,18 +19,19 @@ which are then available in OpenShift to the user.
 
   Scenario Outline: User deploys, checks out and deletes several templates from XpaaS imagestream
    Given Minishift has state "Running"
-    When executing "oc new-app <template-name>" succeeds
+    When executing "oc new-project <project-name>" succeeds
+    Then executing "oc new-app <template-name>" succeeds
```
Comment (agajdosi) [inline test/integration/features/addon-xpaas.feature]: I would suggest `And` here as it is still some kind of preparation.
When I deploy app
Then something works

File: test/integration/features/addon-xpaas.feature [hunk 4]
```diff
@@ -19,18 +19,19 @@ which are then available in OpenShift to the user.
...
      And executing "oc set probe dc/<service-name> --readiness --get-url=http://:8080<http-endpoint>" succeeds
      And service "<service-name>" rollout successfully within "1200" seconds
-    Then with up to "10" retries with wait period of "500ms" the "body" of HTTP request to "<http-endpoint>" of service "<service-name>" in namespace "myproject" contains "<expected-hello>"
-     And with up to "10" retries with wait period of "500ms" the "status code" of HTTP request to "<http-endpoint>" of service "<service-name>" in namespace "myproject" is equal to "200"
-     And executing "oc delete all --all" succeeds
+     And with up to "10" retries with wait period of "500ms" the "body" of HTTP request to "<http-endpoint>" of service "<service-name>" in namespace "<project-name>" contains "<expected-hello>"
```
Comment (agajdosi) [inline test/integration/features/addon-xpaas.feature]: Here I would suggest `Then` as in this moment all preparation steps are done - when I created project and created app from template and set probes and waited for the rollout, then I can do something: check the site, check that and later delete it all.

Comment (amitkrout) [reply test/integration/features/addon-xpaas.feature]: @agajdosi let me know if this is fine for you
```   
   Given Minishift has state "Running"
    When ...
     And ...
     And ...
     And ...
    Then ...
     And ...
     And ...
```

**Comment by amitkrout** [comment]
Addressed all the review comments. 
@LalatenduMohanty This is ready to merge.

**Comment by LalatenduMohanty** [comment]
@amitkrout thanks.

**Comment by budhram** [comment]
Hope this will solve #2692

**Pull request merged**